### ISSUE TITLE
[CLI] segfaults when training

### ISSUE THREAD

**Issue by 903019003** [microsoft/LightGBM#6845]
![Image](https://github.com/user-attachments/assets/2933f2da-99e5-489c-b28e-29bca1d10db4)

## Description

segfaults when training,the log shows that the data has been loaded,but segfaults 
[LightGBM] [Warning] eval_at is set with ndcg_eval_at=1,3,5,10, will be overridden by map_eval_at=1,3,5,10. Current value: eval_at=1,3,5,10
[LightGBM] [Info] Finished loading parameters
[LightGBM] [Info] Using column number 0 as label
[LightGBM] [Info] Calculating query boundaries...
[LightGBM] [Info] Construct bin mappers from text data time 52.62 seconds
[LightGBM] [Info] Making second pass...
[LightGBM] [Info] Total groups: 433515, total data: 11835033
[LightGBM] [Info] Calculating query boundaries...
[LightGBM] [Info] Total groups: 51561, total data: 1410272
[LightGBM] [Info] Finished loading data in 1718.841160 seconds

train.sh: line 4:2374791 $egmentation fault
(core dumped) lightgbm config=train land.conf output model=./models/rank stage2 fuse
Igb ${datadate}.model>./logs/${datadate} land.log

## Environment info

LightGBM version or commit hash:

Command(s) you used to install LightGBM

```shell

```

**Comment by jameslamb** [comment]
Thanks for using LightGBM.

It's not possible for us to help you based only on an error message and a small snippet of logs. If you'd like some help, please provide some of the details that were asked for in the template that came up when you clicked "new issue".

* version of LightGBM
* commands you used to install it
* operating system

And are you able to provide a minimal, reproducible example? ([Stack Overflow docs explaining what that is](https://stackoverflow.com/help/minimal-reproducible-example))

Or at least, can you please provide the **full** set of parameters you used and the **exact** command you used to invoke the CLI?

**Comment by 903019003** [comment]
> Thanks for using LightGBM.
> 
> It's not possible for us to help you based only on an error message and a small snippet of logs. If you'd like some help, please provide some of the details that were asked for in the template that came up when you clicked "new issue".
> 
> * version of LightGBM
> * commands you used to install it
> * operating system
> 
> And are you able to provide a minimal, reproducible example? ([Stack Overflow docs explaining what that is](https://stackoverflow.com/help/minimal-reproducible-example))
> 
> Or at least, can you please provide the **full** set of parameters you used and the **exact** command you used to invoke the CLI?

version of LightGBM:
4.3.0.99
**operating system:**
CentOS Linux release 8.5.2111
Linux version 4.18.0-348.el8.x86_64

**para:**
task = train
boosting_type = gbdt
objective = lambdarank
metric = ndcg,map
ndcg_eval_at = 1,3,5,10
map_eval_at = 1,3,5,10
metric_freq = 1
is_training_metric = true
label_column = 0
max_bin = 255
data = train_data/stage2.train
valid_data = /train_data/stage2.valid
early_stopping = 100
num_trees = 1000
feature_fraction = 1.0
bagging_fraction = 0.8
learning_rate = 0.1
max_depth = 5
num_leaves = 32
two_round=true
num_threads=100
verbosity = 1

**Comment by jameslamb** [comment]
> version of LightGBM:
> 4.3.0.99

That is not a release... it means you're building from some commit between `v4.3.0` and `v4.4.0`. What commit are you building from?

And is it possible to upgrade to the latest release (`v4.6.0`)?

**Comment by jameslamb** [comment]
Please also provide this:

> commands you used to install it

**Comment by 903019003** [comment]
> > version of LightGBM:
> > 4.3.0.99
> 
> That is not a release... it means you're building from some commit between `v4.3.0` and `v4.4.0`. What commit are you building from?
> 
> And is it possible to upgrade to the latest release (`v4.6.0`)?

Thanks for your help :)  !! I found that the problem is caused by the memory being full
